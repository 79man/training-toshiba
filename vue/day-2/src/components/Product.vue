<template>
  <div class="product-card" v-show="renderCondition">
    <img :src="productItem.image" class="product-img" />
    <p class="product-title">{{ productItem.title }}</p>
    <p class="product-price">{{ productItem.price }}</p>
    <button @click.stop="addToCart">Add to Cart</button>
  </div>
</template>
<script>
export default {
  props: {
    productItem: {
      type: Object,
      required: true,
    },
  },
  emits: ['addItem'],
  computed: {
    renderCondition() {
      return (
        this.productItem.category === `men's clothing` ||
        this.productItem.category === 'jewelery'
      )
    },
  },
  methods: {
    addToCart() {
      this.$emit('addItem', this.productItem)
    },
  },
}
</script>
<style lang="scss">
.product-card {
  width: 300px;
  height: 400px;
  border: 1px solid black;
  border-radius: 5px;
  padding: 20px;
  margin: 20px;
  .product-img {
    width: 100%;
    height: 200px;
    object-fit: cover;
  }
  .product-title {
    font-size: 24px;
    text-align: center;
    margin-top: 10px;
  }
  .product-price {
    font-size: 30px;
    font-weight: bold;
    margin-top: 10px;
  }
}
</style>
