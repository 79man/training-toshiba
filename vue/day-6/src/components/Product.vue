<template>
  <div class="product-card">
    <img :src="productItem.image" class="product-img" />
    <p class="product-title">{{ productItem.title }}</p>
    <p class="product-price">{{ productItem.price }}</p>
    <button @click.stop="addToCart">Add to Cart</button>
    <button @click.stop="deleteProduct">Delete</button>
  </div>
</template>
<script>
export default {
  props: {
    productItem: {
      type: Object,
      required: true
    }
  },
  emits: ['addItem', 'deleteItem'],
  methods: {
    addToCart() {
      this.$emit('addItem', this.productItem)
    },
    deleteProduct() {
      this.$emit('deleteItem', this.productItem.id)
    }
  }
}
</script>
<style lang="scss">
.product-card {
  width: 300px;
  height: 400px;
  border: 1px solid black;
  border-radius: 5px;
  padding: 20px;
  margin: 20px;
  .product-img {
    width: 100%;
    height: 200px;
    object-fit: cover;
  }
  .product-title {
    font-size: 24px;
    text-align: center;
    margin-top: 10px;
  }
  .product-price {
    font-size: 30px;
    font-weight: bold;
    margin-top: 10px;
  }
}
</style>
