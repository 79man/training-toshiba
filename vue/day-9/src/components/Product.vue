<template>
  <div class="product-card">
    <img :src="productItem.image" class="product-img" />
    <p class="product-title">{{ upperCase(productItem.title) }}</p>
    <p class="product-price">{{ $n(productItem.price, 'currency') }}</p>
    <button class="add-to-cart" @click.stop="addToCart">Add to Cart</button>
    <button class="delete-item" @click.stop="deleteProduct">Delete</button>
  </div>
</template>
<script>
import { mapActions } from 'vuex'
import { upperCase } from '@/utils/helper'
export default {
  props: {
    productItem: {
      type: Object,
      required: true
    }
  },
  emits: ['deleteItem'],
  methods: {
    ...mapActions('cart', ['setCartList']),
    upperCase,
    addToCart() {
      this.setCartList(this.productItem)
    },
    deleteProduct() {
      this.$emit('deleteItem', this.productItem.id)
    }
  }
}
</script>
<style lang="scss">
.product-card {
  width: 300px;
  height: 400px;
  border: 1px solid black;
  border-radius: 5px;
  padding: 20px;
  margin: 20px;
  .product-img {
    width: 100%;
    height: 200px;
    object-fit: cover;
  }
  .product-title {
    font-size: 24px;
    text-align: center;
    margin-top: 10px;
  }
  .product-price {
    font-size: 30px;
    font-weight: bold;
    margin-top: 10px;
  }
}
</style>
