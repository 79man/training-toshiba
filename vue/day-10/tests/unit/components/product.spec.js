import Product from '@/components/Product'
import { productItem } from '../apiMock/productMock'
import vuexCartMock from '../vuexMock/vuexCartMock'
import { generateWrapper } from '../generate'

describe('Product card tests', () => {
  test('check if product time is rendered correctly in uppercase', () => {
    const wrapper = generateWrapper('shallowMount', Product, {
      props: {
        productItem: productItem
      }
    })
    expect(wrapper.find('.product-title').text()).toBe(
      'FJALLRAVEN - FOLDSACK NO. 1 BACKPACK, FITS 15 LAPTOPS'
    )
  })

  test('check if clicking on addToCart will dispatch an action', async () => {
    const wrapper = generateWrapper('shallowMount', Product, {
      props: {
        productItem: productItem
      }
    })
    wrapper.find('.add-to-cart').trigger('click')
    await wrapper.vm.$nextTick()
    expect(vuexCartMock.actions.setCartList).toHaveBeenCalled()
  })

  test('check if clicking on delete will emit deleteItem', async () => {
    const wrapper = generateWrapper('shallowMount', Product, {
      props: {
        productItem: productItem
      }
    })
    wrapper.find('.delete-item').trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.emitted().deleteItem).toBeTruthy()
  })

  test('check if rendered product card matches snapshot', async () => {
    const wrapper = generateWrapper('shallowMount', Product, {
      props: {
        productItem: productItem
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
