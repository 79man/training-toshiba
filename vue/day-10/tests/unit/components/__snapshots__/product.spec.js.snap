// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product card tests check if rendered product card matches snapshot 1`] = `
<div
  class="product-card"
>
  <img
    class="product-img"
    src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
  />
  <p
    class="product-price"
  />
  <p
    class="product-title"
  >
    FJALLRAVEN - FOLDSACK NO. 1 BACKPACK, FITS 15 LAPTOPS
  </p>
  <button
    class="add-to-cart"
  >
    Add to Cart
  </button>
  <button
    class="delete-item"
  >
    Delete
  </button>
</div>
`;
